llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -1902,15 +1911,6 @@ static Value *getStepVector(Value *Val, Value *Step,
 
   SmallVector<Constant *, 8> Indices;
 
-  // Create a vector of consecutive numbers from zero to VF.
-  VectorType *InitVecValVTy = ValVTy;
-  if (STy->isFloatingPointTy()) {
-    Type *InitVecValSTy =
-        IntegerType::get(STy->getContext(), STy->getScalarSizeInBits());
-    InitVecValVTy = VectorType::get(InitVecValSTy, VLen);
```

Comment:
Perhaps we should assert that `InitVec` is the type we expect, i.e. it must have the same type as `Step`

Reply:
They might not always have the same type as step might be a float or an integer, whereas InitVec is always an integer. Hopefully the calls to `Builder.CreateMul(InitVec, Step)` in the integer case and `Builder.CreateUIToFP(InitVec, ValVTy)` in the float case will catch this.